From 747b0726171496c225a33bd5ae02850071fde707 Mon Sep 17 00:00:00 2001
From: Timothy DeHerrera <tim.deh@pm.me>
Date: Wed, 26 Jan 2022 17:34:54 -0700
Subject: [PATCH] pass in haskell.nix

---
 default.nix | 39 ++++-----------------------------------
 1 file changed, 4 insertions(+), 35 deletions(-)

diff --git a/default.nix b/default.nix
index e592898fb..9f7d8aa5f 100644
--- a/default.nix
+++ b/default.nix
@@ -1,50 +1,20 @@
 { profiling ? false
-, release ? false
+, release ? true
 , threaded ? !profiling
 , checkMaterialization ? false
-
-# Override `src` when this project is imported as a Git submodule:
-#
-# > ttuegel.cleanGitSubtree {
-# >   name = "kore";
-# >   src = ./parent/repo;
-# >   subDir = "path/to/submodule";
-# > };
-#
-# Use `cleanGitSubtree` whenever possible to preserve the same source code
-# layout as the kframework/kore repository (to enable cache re-use).
-#
-, src ? null
+, haskell-nix
+, src
 }:
 
 let
   sources = import ./nix/sources.nix;
-  haskell-nix = import sources."haskell.nix" {};
   inherit (haskell-nix) pkgs;
   inherit (pkgs) lib;
-
-  ttuegel =
-    let
-      src = builtins.fetchGit {
-        url = "https://github.com/ttuegel/nix-lib";
-        rev = "66bb0ab890ff4d828a2dcfc7d5968465d0c7084f";
-        ref = "main";
-      };
-    in import src { inherit pkgs; };
 in
 
 let
   project = pkgs.haskell-nix.stackProject {
-    src = ttuegel.cleanSourceWith {
-      name = "kore";
-      src = ttuegel.orElse src (ttuegel.cleanGitSubtree { src = ./.; });
-      ignore = [
-        "/*"
-        "!/stack.yaml"
-        "!/kore"
-      ];
-    };
-    inherit checkMaterialization;
+    inherit checkMaterialization src;
     materialized = ./nix/kore.nix.d;
     modules = [
       {
@@ -72,7 +42,6 @@ let
   shell = import ./shell.nix { inherit default checkMaterialization; };
 
   version = project.kore.components.exes.kore-exec.version;
-  # version = "0.43.0.0";
 
   prelude-kore = ./src/main/kore/prelude.kore;
 
-- 
2.34.1

