diff --git a/Makefile b/Makefile
index 06d8186d..5b098dbc 100644
--- a/Makefile
+++ b/Makefile
@@ -6,6 +6,9 @@ UNAME_S := $(shell uname -s)
 DEPS_DIR      := deps
 BUILD_DIR     := .build
 NODE_DIR      := $(abspath node)
+SUBDEFN_DIR   := .
+DEFN_BASE_DIR := $(BUILD_DIR)/defn
+DEFN_DIR      := $(DEFN_BASE_DIR)/$(SUBDEFN_DIR)
 BUILD_LOCAL   := $(abspath $(BUILD_DIR)/local)
 LOCAL_LIB     := $(BUILD_LOCAL)/lib
 LOCAL_BIN     := $(BUILD_LOCAL)/bin
@@ -46,7 +49,7 @@ export PLUGIN_SUBMODULE
 
 .PHONY: all clean distclean                                                                                                      \
         deps k-deps plugin-deps libsecp256k1 libff protobuf                                                                      \
-        build build-haskell build-llvm build-provex build-node build-kevm                                                        \
+        build build-haskell build-llvm build-provex build-vm build-kevm                                                        \
         test test-all test-conformance test-rest-conformance test-all-conformance test-slow-conformance test-failing-conformance \
         test-vm test-rest-vm test-all-vm test-bchain test-rest-bchain test-all-bchain test-node                                  \
         test-prove test-failing-prove                                                                                            \
@@ -70,16 +73,15 @@ distclean:
 # Non-K Dependencies
 # ------------------
 
-libsecp256k1_out := $(LOCAL_LIB)/pkgconfig/libsecp256k1.pc
-libff_out        := $(KEVM_LIB)/libff/lib/libff.a
-libcryptopp_out  := $(KEVM_LIB)/cryptopp/lib/libcryptopp.a
 protobuf_out     := $(LOCAL_LIB)/proto/proto/msg.pb.cc
 
-libsecp256k1: $(libsecp256k1_out)
-libff:        $(libff_out)
-libcryptopp : $(libcryptopp_out)
 protobuf:     $(protobuf_out)
 
+ifndef SYSTEM_LIBSECP256K1
+
+libsecp256k1_out := $(LOCAL_LIB)/pkgconfig/libsecp256k1.pc
+libsecp256k1: $(libsecp256k1_out)
+
 $(libsecp256k1_out): $(PLUGIN_SUBMODULE)/deps/secp256k1/autogen.sh
 	cd $(PLUGIN_SUBMODULE)/deps/secp256k1                                 \
 	    && ./autogen.sh                                                   \
@@ -87,6 +89,13 @@ $(libsecp256k1_out): $(PLUGIN_SUBMODULE)/deps/secp256k1/autogen.sh
 	    && $(MAKE)                                                        \
 	    && $(MAKE) install
 
+endif # ifndef SYSTEM_LIBSECP256K1
+
+ifndef SYSTEM_LIBFF
+
+libff_out := $(KEVM_LIB)/libff/lib/libff.a
+libff: $(libff_out)
+
 LIBFF_CMAKE_FLAGS :=
 
 ifeq ($(UNAME_S),Linux)
@@ -103,14 +112,20 @@ $(libff_out): $(PLUGIN_SUBMODULE)/deps/libff/CMakeLists.txt
 	    && cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$(INSTALL_LIB)/libff $(LIBFF_CMAKE_FLAGS) \
 	    && make -s -j4                                                                                          \
 	    && make install DESTDIR=$(CURDIR)/$(BUILD_DIR)
+endif # ifndef SYSTEM_LIBFF
 
 $(protobuf_out): $(NODE_DIR)/proto/msg.proto
 	@mkdir -p $(LOCAL_LIB)/proto
 	protoc --cpp_out=$(LOCAL_LIB)/proto -I $(NODE_DIR) $(NODE_DIR)/proto/msg.proto
 
+ifndef SYSTEM_LIBCRYPTOPP
+libcryptopp_out  := $(KEVM_LIB)/cryptopp/lib/libcryptopp.a
+libcryptopp : $(libcryptopp_out)
+
 $(libcryptopp_out): $(PLUGIN_SUBMODULE)/deps/cryptopp/GNUmakefile
 	cd $(PLUGIN_SUBMODULE)/deps/cryptopp                            \
             && $(MAKE) install DESTDIR=$(CURDIR)/$(BUILD_DIR) PREFIX=$(INSTALL_LIB)/cryptopp
+endif # SYSTEM_LIBCRYPTOPP
 
 # K Dependencies
 # --------------
@@ -312,7 +327,7 @@ build: $(patsubst %, $(KEVM_BIN)/%, $(install_bins)) $(patsubst %, $(KEVM_LIB)/%
 
 build-llvm:    $(KEVM_LIB)/$(llvm_kompiled)    $(KEVM_LIB)/kore-json.py
 build-haskell: $(KEVM_LIB)/$(haskell_kompiled) $(KEVM_LIB)/kore-json.py
-build-node:    $(KEVM_LIB)/$(node_kompiled)
+build-vm:      $(KEVM_LIB)/$(node_kompiled)
 build-kevm:    $(KEVM_BIN)/kevm $(kevm_includes) $(lemma_includes) $(plugin_includes)
 
 all_bin_sources := $(shell find $(KEVM_BIN) -type f | sed 's|^$(KEVM_BIN)/||')
diff --git a/cmake/node/CMakeLists.txt b/cmake/node/CMakeLists.txt
index a9f695c0..5289f4f3 100644
--- a/cmake/node/CMakeLists.txt
+++ b/cmake/node/CMakeLists.txt
@@ -4,8 +4,14 @@ execute_process(COMMAND which kompile OUTPUT_VARIABLE KOMPILE_PATH
     OUTPUT_STRIP_TRAILING_WHITESPACE)
 execute_process(COMMAND dirname ${KOMPILE_PATH} OUTPUT_VARIABLE K_BIN
     OUTPUT_STRIP_TRAILING_WHITESPACE)
+execute_process(COMMAND which llvm-kompile OUTPUT_VARIABLE LLVM_PATH
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+execute_process(COMMAND dirname ${LLVM_PATH} OUTPUT_VARIABLE LLVM_BIN
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
 set(K_LIB ${K_BIN}/../lib)
+set(LLVM_LIB ${LLVM_BIN}/../lib)
 list(APPEND CMAKE_MODULE_PATH "${K_LIB}/cmake/kframework")
+list(APPEND CMAKE_MODULE_PATH "${LLVM_LIB}/cmake/kframework")
 include(LLVMKompilePrelude)
 project (KevmVm CXX)
 
@@ -39,11 +45,11 @@ target_include_directories(kevm-vm
 	PUBLIC $ENV{NODE_DIR}/vm/kevm
 	PUBLIC ${CMAKE_SOURCE_DIR}/..
 	PUBLIC $ENV{KEVM_LIB_ABS}/libff/include
-	PUBLIC ${K_LIB}/../include/kllvm)
+    PUBLIC ${K_LIB}/../include/kllvm
+    PUBLIC ${LLVM_BIN}/../include/kllvm)
 target_compile_options(kevm-vm
 	PUBLIC $ENV{LLVM_KOMPILE_OPTS}
-	PUBLIC -Wall -Werror)
+    PUBLIC -Wall -Werror -Wno-unused-command-line-argument -Wno-unused-result)
 target_link_libraries(kevm-vm
 	ff protobuf gmp ${LINK_PROCPS} cryptopp secp256k1 crypto)
-
 include(LLVMKompile)
